File: Blackjack.py
import random

def draw():
    x = random.choice(current_deck)
    current_deck.remove(x)
    return x

def count_total(x):
    total = 0
    for card in x:
        if card == 'J' or card == 'Q' or card == 'K':
            card = 10
        if card == 'A':
            card = 1
        total += int(card)
    return total

player_total = 0
dealer_total = 0
keep_playing = 'yes'
money = input('Enter starting money. >> ')
money = float(money.replace('$',''))

while keep_playing.lower() != 'no' and keep_playing.lower() != 'n':
    current_deck = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A'] * 4
    player_hand = []
    dealer_hand = []
    bet_amount = input('Enter bet amount. >> ')
    bet_amount = float(bet_amount.replace('$',''))
    player_hand.append(draw())
    player_hand.append(draw())
    print ('Player has ' + str(player_hand))
    player_total = count_total(player_hand)
    print ('Player point total is ' + str(player_total), end = '')
    if 'A' in player_hand:
        print(' or ' + str(player_total+10))
    else:
        print('')
    dealer_hand.append(draw())
    print ('Dealer has ' + str(dealer_hand))
    dealer_total = count_total(dealer_hand)
    print ('Dealer point total is ' + str(dealer_total), end = '')
    if 'A' in dealer_hand:
        print(' or ' + str(dealer_total+10))
    else:
        print('')
    stop = False
    action = ''
    while player_total < 21 and stop == False:
        action = input('Hit or stay? ')
        if action.lower() == 'hit' or action.lower() == 'h':
            player_hand.append(draw())
            print ('Player has ' + str(player_hand))
            player_total = count_total(player_hand)
            print ('Player point total is ' + str(player_total), end = '')
            if 'A' in player_hand:
                print(' or ' + str(player_total+10))
            else:
                print('')
        else:
            stop = True
    if player_total > 21:
        print ('Player busts, you lose')
        money -= bet_amount
    else:
        while dealer_total < 17 or (dealer_total + 10 < 17 and 'A' in dealer_hand):
            dealer_hand.append(draw())
            dealer_total = count_total(dealer_hand)
            print ('Dealer has ' + str(dealer_hand))
            print ('Dealer point total is ' + str(dealer_total), end = '')
            if 'A' in dealer_hand:
                print(' or ' + str(dealer_total+10))
            else:
                print('')
        if 'A' in player_hand and player_total + 10 <= 21:
            player_total += 10
        if 'A' in dealer_hand and dealer_total + 10 <= 21:
            dealer_total += 10
        if player_total > dealer_total:
                print ('You win')
                money += bet_amount
        if player_total < dealer_total and dealer_total <= 21:
                print ('You lose')
                money -= bet_amount
        if player_total == dealer_total:
                print ('Push')
        if dealer_total > 21:
                print ('Dealer busts, you win')
                money += bet_amount
    print('You have $' + str(money) + '.')
    keep_playing = input('Keep playing? ')

File: Blackjack.js
function hit() {
    draw_cards(player_hand, "player_hand")
    console.log("Current Deck: " + current_deck);
    console.log("Player Hand: " + player_hand);
    player_total = total_hand(player_hand)
    console.log("Player Total: " + player_total);
    if (player_total > 21) {
        document.getElementById("status").value = "You Bust, You Lose";
        current_money -= bet
        document.getElementById("current_money").value = current_money;
        console.log("Current Money: " + current_money);
        setup_game()
    }
}
function stay() {
    player_total = total_hand(player_hand)
    while (dealer_total <= 16) {
        draw_cards(dealer_hand, "dealer_hand");
        dealer_total = total_hand(dealer_hand);
        console.log("Dealer Hand: " + dealer_hand);
        console.log("Dealer Total: " + dealer_total);
    }
    if (dealer_total > 21) {
        document.getElementById("status").value = "Dealer Busts, You Win";
        current_money += bet
        document.getElementById("current_money").value = current_money;

    }
    else if (player_total == dealer_total) {
        document.getElementById("status").value = "Push";
    }
    else if (player_total > dealer_total) {
        document.getElementById("status").value = "You Win";
        current_money += bet
        document.getElementById("current_money").value = current_money;
    }
    else if (player_total < dealer_total) {
        document.getElementById("status").value = "You Lose";
        current_money -= bet
        document.getElementById("current_money").value = current_money;
    }
    console.log("Current Money: " + current_money);
    setup_game()
}

function new_bet() {
    document.getElementById("dealer_hand").value = [];
    document.getElementById("player_hand").value = [];
    document.getElementById('bet_amount').readOnly = true;
    document.getElementById("dealer_hand").value = null;
    bet = parseFloat(document.getElementById("bet_amount").value);
    console.log("Bet Amount: " + bet);
    for (x = 0; x <= 1; x++) {
        draw_cards(player_hand, "player_hand")
    }
    draw_cards(dealer_hand, "dealer_hand")
    player_total = total_hand(player_hand);
    dealer_total = total_hand(dealer_hand);
    console.log("Current Deck: " + current_deck);
    console.log("Player Hand: " + player_hand);
    console.log("Player Total: " + player_total);
    console.log("Dealer Hand: " + dealer_hand);
    console.log("Dealer Total: " + dealer_total);
    document.getElementById("hit_button").disabled = false;
    document.getElementById("stay_button").disabled = false;
    document.getElementById("submit_bet").disabled = true;
    document.getElementById("status").value = "Hit or Stay";
}

function draw_cards(hand, element) {
    let drawn = current_deck[Math.floor(Math.random() * current_deck.length)];
    hand.push(drawn);
    current_deck.pop(current_deck.indexOf(drawn));
    document.getElementById(element).value += " " + drawn;
}

function total_hand(hand) {
    let total = 0
    for (var i = 0; i < hand.length; i++) {
        card = hand[i]
        if (card === "K" || card == "Q" || card == "J" || card == "10") { total += 10; }
        if (card === "9") { total += 9; }
        if (card === "8") { total += 8; }
        if (card === "7") { total += 7; }
        if (card === "6") { total += 6; }
        if (card === "5") { total += 5; }
        if (card === "4") { total += 4; }
        if (card === "3") { total += 3; }
        if (card === "2") { total += 2; }
        if (card === "A") { total += 1; }
    }
    if (hand.includes("A") && total <= 11) { total += 10; }
    return total
}

function setup_game() {
    console.log("Reset");
    current_deck = deck
    bet = 0;
    dealer_hand = []
    player_hand = []
    dealer_total = 0
    player_total = 0
    document.getElementById('bet_amount').readOnly = false;
    document.getElementById("bet_amount").value = null;
    document.getElementById("submit_bet").disabled = false;
    document.getElementById("hit_button").disabled = true;
    document.getElementById("stay_button").disabled = true;
}
const deck = ["A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2",
    "A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2",
    "A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2",
    "A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2"];
let current_deck = deck
let bet = 0;
let dealer_hand = []
let player_hand = []
let dealer_total = 0
let player_total = 0
let current_money = 10000
document.getElementById("hit_button").disabled = true;
document.getElementById("stay_button").disabled = true;

console.log("Current Deck: " + current_deck);
console.log("Player Hand: " + player_hand);
console.log("Dealer Hand: " + dealer_hand);
console.log("Player Total: " + player_total);
console.log("Dealer Total: " + dealer_total);
console.log("Current Money: " + current_money);
